packages/ember-toucan-core/src/-private/assert-block-or-argument-exists.ts

```diff
@@ -0,0 +1,39 @@
+import { assert } from '@ember/debug';
+
+export type AssertBlockOrArg = {
+  blockExists: boolean;
+  argName: string;
+  arg?: string;
+  isRequired?: boolean;
+};
+
+/**
+ * @param {object} AssertBlockOrArg
+ *
+ **/
+const assertBlockOrArgumentExists = ({
+  blockExists,
+  argName,
+  arg,
+  isRequired,
+}: AssertBlockOrArg) => {
+  if (isRequired) {
+    assert(
+      `You need either :${argName} or @${argName}`,
```

Comment:
question: Do you think we should put `@${argName}` first to prompt folks to use the argument rather than the named block?  Or do you think more people will use the named block version?